// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package dsidRpc;

// The greeting service definition.
service Dsid {
  // Sends a greeting
  rpc voidCall (Empty) returns (Empty) {}

  rpc longCall (longRequest) returns (longReply) {}

  rpc eightLongCall(eightLongRequest) returns(eightLongReply) {}

  rpc stringCall(stringRequest) returns (stringReply) {}

  rpc complexCall(complexRequest) returns (complexReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message Empty {

}

message longRequest {
    int64 numberRequest = 1;
}
message longReply {
    int64 numberReply = 1;
}
message eightLongRequest {
    repeated int64 arrayRequest = 1;
}
message eightLongReply {
    repeated int64 arrayReply = 1;
}
message stringRequest {
    string nameRequest = 1;
}
message stringReply {
    string nameReply = 1;
}
message complexRequest {
    bytes alunoRequest = 1;
}
message complexReply {
    bytes alunoReply = 1;
}


